bindings/matrix-sdk-ffi/src/client.rs

```diff
@@ -54,18 +60,34 @@ impl Client {
     }
 
     /// Login using a username and password.
-    pub fn login(&self, username: String, password: String) -> anyhow::Result<()> {
+    pub fn login(
+        &self,
+        username: String,
+        password: String,
+        initial_device_name: String,
+        device_id: Option<String>,
+    ) -> anyhow::Result<()> {
```

Comment:
starting to add all possible parameters begs the question if we shouldn't just expose the `LoginBuilder`... but probably not as part of this PR.

in any case, I think the `initial_device_name` should be optional, too...

```suggestion
        &self,
        username: String,
        password: String,
        initial_device_name: Option<String>,
        device_id: Option<String>,
    ) -> anyhow::Result<()> {
```